common/lib/xmodule/xmodule/tests/test_content.py

```diff
@@ -173,3 +179,101 @@ def test_static_content_write_js(self):
+
+            # Create a locked image.
+            locked_buf = StringIO()
+            locked_name = "o_l.png"
+
+            # Save the course image to the content store.
+            locked_asset_key = StaticContent.compute_location(self.old_course.id, locked_name)
+            locked_content = StaticContent(locked_asset_key, locked_name, 'image/png', locked_buf, locked=True)
+            contentstore().save(locked_content)
+
+    @ddt.data(
+        (None, u"n_unl.png", u"/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (None, u"n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (u"dev", u"n_unl.png", u"//dev/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (u"dev", u"n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (None, u"/n_unl.png", u"/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (None, u"/n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (u"dev", u"/n_unl.png", u"//dev/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (u"dev", u"/n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (None, u"/asset-v1:a+b+d+type@asset+block@n_unl.png", u"/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (None, u"/asset-v1:a+b+d+type@asset+block@n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (u"dev", u"/asset-v1:a+b+d+type@asset+block@n_unl.png", u"//dev/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (u"dev", u"/asset-v1:a+b+d+type@asset+block@n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (u"dev", u"/c4x/a/b/asset/n_unl.png", u"//dev/c4x/a/b/asset/n_unl.png"),
+        (u"dev", u"/c4x/a/b/asset/n_l.png", u"//dev/c4x/a/b/asset/n_l.png"),
+    )
+    @ddt.unpack
+    def test_canonical_asset_path_with_new_style_assets(self, base_url, asset_name, expected_path):
+        StaticContent.base_url = base_url
+
+        asset_path = StaticContent.get_canonicalized_asset_path(self.new_course.id, asset_name)
+        self.assertEqual(asset_path, expected_path)
+
+    @ddt.data(
+        (None, u"o_unl.png", u"/c4x/a/c/asset/o_unl.png"),
+        (None, u"o_l.png", u"/c4x/a/c/asset/o_l.png"),
+        (u"dev", u"o_unl.png", u"//dev/c4x/a/c/asset/o_unl.png"),
+        (u"dev", u"o_l.png", u"/c4x/a/c/asset/o_l.png"),
+        (None, u"/o_unl.png", u"/c4x/a/c/asset/o_unl.png"),
+        (None, u"/o_l.png", u"/c4x/a/c/asset/o_l.png"),
+        (u"dev", u"/o_unl.png", u"//dev/c4x/a/c/asset/o_unl.png"),
+        (u"dev", u"/o_l.png", u"/c4x/a/c/asset/o_l.png"),
+        (None, u"/c4x/a/c/asset/o_unl.png", u"/c4x/a/c/asset/o_unl.png"),
+        (None, u"/c4x/a/c/asset/o_l.png", u"/c4x/a/c/asset/o_l.png"),
+        (u"dev", u"/c4x/a/c/asset/o_unl.png", u"//dev/c4x/a/c/asset/o_unl.png"),
+        (u"dev", u"/c4x/a/c/asset/o_l.png", u"/c4x/a/c/asset/o_l.png"),
+    )
+    @ddt.unpack
+    def test_canonical_asset_path_with_c4x_style_assets(self, base_url, asset_name, expected_path):
+        StaticContent.base_url = base_url
```

Comment:
On second thought, it might be best to actually just mock that value within this test, so that we don't have to worry about the cleanup across tests within the test case as well.